src/pages/workspace/taxes/WorkspaceTaxesPage.tsx

```diff
@@ -0,0 +1,125 @@
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import withPolicyAndFullscreenLoading from '@pages/workspace/withPolicyAndFullscreenLoading';
+import type {WithPolicyAndFullscreenLoadingProps} from '@pages/workspace/withPolicyAndFullscreenLoading';
+import type SCREENS from '@src/SCREENS';
+
+type PolicyForList = {
+    value: string;
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+    rightElement: React.ReactNode;
+};
+
+type WorkspaceTaxesPageProps = WithPolicyAndFullscreenLoadingProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.TAXES>;
+
+function WorkspaceTaxesPage({policy}: WorkspaceTaxesPageProps) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+    const [selectedTaxes, setSelectedTaxes] = useState<string[]>([]);
+
+    const taxesList = useMemo<PolicyForList[]>(
+        () =>
+            Object.entries(policy?.taxRates?.taxes ?? {}).map(([key, value]) => ({
+                // TODO: Clean up: check if all properties are needed
+                value: value.name,
+                text: value.name,
+                keyForList: key,
+                isSelected: !!selectedTaxes.includes(key),
+                rightElement: (
+                    // TODO: Extract this into a separate component together with WorkspaceCategoriesPage
+                    <View style={styles.flexRow}>
+                        <Text style={[styles.disabledText, styles.alignSelfCenter]}>{value.isDisabled ? translate('workspace.common.disabled') : translate('workspace.common.enabled')}</Text>
+                        <View style={[styles.p1, styles.pl2]}>
+                            <Icon
+                                src={Expensicons.ArrowRight}
+                                fill={theme.icon}
+                            />
+                        </View>
+                    </View>
+                ),
+            })),
+        [policy?.taxRates?.taxes, selectedTaxes, styles, theme.icon, translate],
+    );
+
+    const toggleTax = (tax: PolicyForList) => {
+        setSelectedTaxes((prev) => {
+            if (prev.includes(tax.keyForList)) {
+                return prev.filter((item) => item !== tax.keyForList);
+            }
+            return [...prev, tax.keyForList];
+        });
+    };
+
+    const toggleAllTaxes = () => {
+        const isAllSelected = selectedTaxes.length === taxesList.length;
+        if (isAllSelected) {
+            setSelectedTaxes([]);
+        } else {
+            setSelectedTaxes(taxesList.map((item) => item.keyForList));
+        }
```

Comment:
We should use a similar approach to the `toggleTax`. It should be safer.

```suggestion
     setSelectedTaxes((prev) => {
        if (prev.length === taxesList.length) {
            return [];
        } else {
            return taxesList.map((item) => item.keyForList);
        }
    });
```